<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>739</width>
    <height>660</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>VSL</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>739</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionnew"/>
   <addaction name="separator"/>
   <addaction name="actionplay"/>
   <addaction name="separator"/>
   <addaction name="actionopenVSL"/>
   <addaction name="actionsave_VSL"/>
   <addaction name="actionexport"/>
   <addaction name="actionpaste_ss"/>
   <addaction name="actionrandom"/>
   <addaction name="actionopenetg"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QDockWidget" name="dock_editor">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>editor</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1,0">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>5</number>
     </property>
     <property name="topMargin">
      <number>5</number>
     </property>
     <property name="rightMargin">
      <number>5</number>
     </property>
     <property name="bottomMargin">
      <number>5</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,0,0,0,0,0">
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="symbols">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="title">
          <string/>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0,0,0,0,0,0" columnstretch="0,0,0,0,0,0">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="2">
           <widget class="QPushButton" name="pushButton">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>π</string>
            </property>
           </widget>
          </item>
          <item row="6" column="2">
           <widget class="QPushButton" name="pushButton_13">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>☯</string>
            </property>
           </widget>
          </item>
          <item row="5" column="5">
           <widget class="QPushButton" name="pushButton_9">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>♪</string>
            </property>
           </widget>
          </item>
          <item row="6" column="4">
           <widget class="QPushButton" name="pushButton_15">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>✬</string>
            </property>
           </widget>
          </item>
          <item row="6" column="3">
           <widget class="QPushButton" name="pushButton_14">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>§</string>
            </property>
           </widget>
          </item>
          <item row="5" column="3">
           <widget class="QPushButton" name="pushButton_11">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>∿</string>
            </property>
           </widget>
          </item>
          <item row="6" column="5">
           <widget class="QPushButton" name="pushButton_16">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>➡</string>
            </property>
           </widget>
          </item>
          <item row="5" column="4">
           <widget class="QPushButton" name="pushButton_12">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>τ</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QPushButton" name="pushButton_4">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>[]</string>
            </property>
           </widget>
          </item>
          <item row="0" column="5">
           <widget class="QPushButton" name="pushButton_8">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>\\</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QPushButton" name="pushButton_6">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>||</string>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <widget class="QPushButton" name="pushButton_10">
            <property name="minimumSize">
             <size>
              <width>32</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>⬳</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="pushButton_2">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Ø</string>
            </property>
           </widget>
          </item>
          <item row="2" column="5">
           <widget class="QPushButton" name="pushButton_7">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="text">
             <string>‹›</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QPushButton" name="pushButton_5">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>()</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QPushButton" name="pushButton_3">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>{}</string>
            </property>
           </widget>
          </item>
          <item row="7" column="2">
           <widget class="QPushButton" name="pushButton_17">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>♯</string>
            </property>
           </widget>
          </item>
          <item row="7" column="3">
           <widget class="QPushButton" name="pushButton_18">
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>♭</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="func_list">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;insert funcion in formula&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chb_decompile">
         <property name="text">
          <string>view llvm ir-code</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTableWidget" name="waveTable">
         <property name="maximumSize">
          <size>
           <width>150</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="columnCount">
          <number>1</number>
         </property>
         <attribute name="horizontalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <row>
          <property name="text">
           <string>channels</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>sample rate</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>bits sample</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>seconds</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string># samples</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>volume %</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>lap (ms)</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>samps/sec</string>
          </property>
         </row>
         <column/>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="progress">
         <property name="font">
          <font>
           <family>Courier New</family>
           <pointsize>18</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">color:red; background-color: yellow; border: none; </string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1">
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="CodeEditor" name="editor">
           <property name="font">
            <font>
             <family>Courier New</family>
            </font>
           </property>
           <property name="plainText">
            <string>~440;
~442;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <family>Courier New</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string> replace </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="replace_org">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;replace this text&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="statusTip">
            <string>replace</string>
           </property>
           <property name="whatsThis">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <family>Courier New</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string> with </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="replace_with">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;with...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="statusTip">
            <string>whit...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="replace_button">
           <property name="font">
            <font>
             <family>Courier New</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>replace</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QListWidget" name="listFiles">
         <property name="maximumSize">
          <size>
           <width>150</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="decompiled">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Courier New</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="acceptDrops">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dock_graph">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>graph</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>5</number>
     </property>
     <property name="topMargin">
      <number>5</number>
     </property>
     <property name="rightMargin">
      <number>5</number>
     </property>
     <property name="bottomMargin">
      <number>5</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0">
       <property name="rightMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="WaveGraph" name="wave_graph_widget" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="textVisible">
          <bool>true</bool>
         </property>
         <property name="invertedAppearance">
          <bool>false</bool>
         </property>
         <property name="textDirection">
          <enum>QProgressBar::TopToBottom</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionnew">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/new.png</normaloff>:/vs/icons/new.png</iconset>
   </property>
   <property name="text">
    <string>new</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionplay">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/play.png</normaloff>
     <normalon>:/vs/icons/pause.png</normalon>:/vs/icons/play.png</iconset>
   </property>
   <property name="text">
    <string>play</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionexport">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/export1.png</normaloff>:/vs/icons/export1.png</iconset>
   </property>
   <property name="text">
    <string>export</string>
   </property>
  </action>
  <action name="actionpaste_ss">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/clippaste.ico</normaloff>:/vs/icons/clippaste.ico</iconset>
   </property>
   <property name="text">
    <string>paste_ss</string>
   </property>
   <property name="toolTip">
    <string>paste from spreadsheet</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+V</string>
   </property>
  </action>
  <action name="actionrandom">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/rand.png</normaloff>:/vs/icons/rand.png</iconset>
   </property>
   <property name="text">
    <string>random</string>
   </property>
  </action>
  <action name="actionsave_VSL">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/save.png</normaloff>:/vs/icons/save.png</iconset>
   </property>
   <property name="text">
    <string>saveVSL</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionopenVSL">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/open.png</normaloff>:/vs/icons/open.png</iconset>
   </property>
   <property name="text">
    <string>openVSL</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionopenetg">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/vs/icons/transformation.png</normaloff>:/vs/icons/transformation.png</iconset>
   </property>
   <property name="text">
    <string>openetg</string>
   </property>
   <property name="toolTip">
    <string>open old etg file</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WaveGraph</class>
   <extends>QWidget</extends>
   <header>wavegraph.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">codeeditor.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>editor</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>pushButton_12</tabstop>
  <tabstop>pushButton_15</tabstop>
  <tabstop>pushButton_13</tabstop>
  <tabstop>pushButton_14</tabstop>
  <tabstop>pushButton_6</tabstop>
  <tabstop>pushButton_8</tabstop>
  <tabstop>pushButton_5</tabstop>
  <tabstop>pushButton_7</tabstop>
  <tabstop>pushButton_9</tabstop>
  <tabstop>pushButton_16</tabstop>
  <tabstop>func_list</tabstop>
  <tabstop>pushButton_4</tabstop>
  <tabstop>pushButton_11</tabstop>
  <tabstop>pushButton_2</tabstop>
  <tabstop>pushButton_3</tabstop>
  <tabstop>replace_org</tabstop>
  <tabstop>replace_with</tabstop>
  <tabstop>replace_button</tabstop>
  <tabstop>pushButton_17</tabstop>
  <tabstop>pushButton_18</tabstop>
  <tabstop>chb_decompile</tabstop>
  <tabstop>waveTable</tabstop>
  <tabstop>listFiles</tabstop>
  <tabstop>decompiled</tabstop>
  <tabstop>pushButton_10</tabstop>
 </tabstops>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
